code/controllers/subsystems/overlays.dm

```diff
@@ -1,242 +1,239 @@
-	if (!length(add))
-		return
-	if (priority)
-		if (priority_overlays)
-			priority_overlays += add
-		else
-			priority_overlays = add
-	else if (our_overlays)
-		our_overlays += add
-	else
-		our_overlays = add
-	QueueOverlayUpdate()
-
-
-/**
- * Removes specific overlay(s) from the atom.  Usually does not remove them from "priority" overlays.
- *
- * @param overlays The overlays to removed, type can be anything that is allowed for add_overlay().
- * @param priority If true, also will remove them from the "priority" overlays.
- */
-/atom/proc/cut_overlay(list/cut, priority)
-	if (!cut)
-		return
-	cut = SSoverlays.GetAppearanceList(src, cut)
-	if (!length(cut))
-		return
-	var/update
-	if (priority && length(priority_overlays))
-		priority_overlays -= cut
-		update = TRUE
-	if (length(our_overlays))
-		our_overlays -= cut
-		update = TRUE
-	if (update)
-		QueueOverlayUpdate()
-
-
-/**
- * Copy the overlays from another atom, either replacing all of ours or appending to our existing overlays.
- * Note: This copies only the normal overlays, not the "priority" overlays.
- *
- * @param other The atom to copy overlays from.
- * @param cut_old If true, all of our overlays will be *replaced* by the other's. If other is null, that means cutting all ours.
- */
-/atom/proc/copy_overlays(atom/other, cut)
-	if (!other)
-		if (cut)
-			cut_overlays()
-		return
-	if (!length(other.our_overlays))
-		if (cut)
-			cut_overlays()
-		return
-	if (cut || !length(our_overlays))
-		our_overlays = other.our_overlays.Copy()
-	else
-		our_overlays |= other.our_overlays
-	QueueOverlayUpdate()
-
-
-/**
- * Returns a list of overlays that the target atom has
- *
- * @param priority If true, returns priority overlays as well
- * @param special If true, returns special overlays like emissives and em_blockers
- */
-/proc/get_overlays(atom/other, priority, special)
-	var/list/including = list()
-	if (!other)
-		return including
-	for (var/image/I as anything in other.our_overlays)
-		if (!special && I.plane > 0)
-			continue
-		including += I
-	if (!priority)
-		return including
-	for (var/image/I as anything in other.priority_overlays)
-		if (!special && I.plane > 0)
-			continue
-		including += I
-	return including
+SUBSYSTEM_DEF(overlays)
+	name = "Overlay"
+	flags = SS_TICKER
+	wait = 1 // SS_TICKER - Ticks
+	priority = FIRE_PRIORITY_OVERLAYS
+	init_order = INIT_ORDER_OVERLAY
+
+	/// The queue of atoms that need overlay updates.
+	var/static/tmp/list/queue = list()
+
+	/// A list([icon] = list([state] = [appearance], ...), ...) cache of appearances.
+	var/static/tmp/list/state_cache = list()
+
+	/// A list([icon] = [appearance], ...) cache of appearances.
+	var/static/tmp/list/icon_cache = list()
+
+	/// The number of appearances currently cached.
+	var/static/tmp/cache_size = 0
+
+
+/datum/controller/subsystem/overlays/Recover()
+	queue.Cut()
+	state_cache.Cut()
+	icon_cache.Cut()
+	cache_size = 0
+	for (var/atom/atom)
+		atom.flags &= ~OVERLAY_QUEUED
+		CHECK_TICK

```

Comment:
adds this

Reply:
Recover shouldn't sleep afaik, it's better to lag than risk something interacting with the subsystem while it's recovering.